initialization/initialization.go

```diff
@@ -295,7 +307,7 @@ func (init *Initializer) Initialize(ctx context.Context) error {
 		zap.Int("firstFileIndex", layout.FirstFileIdx),
 		zap.Int("lastFileIndex", layout.LastFileIdx),
 	)
-	if err := removeRedundantFiles(init.cfg, init.opts, init.logger); err != nil {
+	if err := checkRedundantFiles(init.cfg, init.opts, init.keepRedundantFiles, init.logger); err != nil {
 		return err
 	}
```

Comment:
Instead of passing the `bool` to the function I'd prefer to instead not call `removeRedundantFiles` when it is `true`:

```go
if !init.keepRedundantFiles {
	if err := removeRedundantFiles(init.cfg, init.opts, init.logger); err != nil {
		return err
	}
}
```

no need to iterate over all files telling the user what is all ignored when the user actively chose to ignore files.

Reply:
Certainly fine as well.

But in the usage scenarios I envision, many users will set "keep" as a default everywhere, and then still find it helpful to see log messages where additional files are discovered, to assist them in keeping track of things. The log message helps in cases where one forgot to adjust the num-units after increase; it also helps in cleaning up after decrease: it basically reminds the user to think about a situation that was previously handled in a one-sided manner by the program with deleting files.

In fact, I originally had exactly the variant you propose, with no log message at all :) I only added the log message because I thought that might make it easier to upstream. Having used it while resizing (mostly increasing) lots and lots of postdata over the past few weeks, I have come to slightly prefer the log message.

All that said: either way is fine with me.